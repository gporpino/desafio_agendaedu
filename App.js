import React, { Component } from 'react';
import Routes from './src/routes';
import { AppLoading, Asset, Font } from 'expo';

// function cacheImages(images) {
//   return images.map((image) => {
//     if (typeof image === 'string') {
//       return Image.prefetch(image);
//     }
//     return Asset.fromModule(image).downloadAsync();
//   });
// }

function cacheFonts(fonts) {
  return fonts.map(font => Font.loadAsync(font));
}

export default class App extends Component {
  state = {
    isReady: false
  };

  loadAssetsAsync = async () => {
    console.log('start');
    await Font.loadAsync({
      'sf-pro-display-regular': require('./assets/fonts/sf-pro-display-regular.otf'),
      'sf-pro-display-bold': require('./assets/fonts/sf-pro-display-bold.otf'),
      'sf-pro-display-semibold': require('./assets/fonts/sf-pro-display-semibold.otf')
    });

    this.setState({ fontLoaded: true });
  };

  finishLoading = () => {
    console.log('ends');
    this.setState({ isReady: true });
  };

  render() {
    console.log('render');
    if (!this.state.isReady) {
      return (
        <AppLoading
          startAsync={this.loadAssetsAsync}
          onFinish={this.finishLoading}
        />
      );
    } else {
      console.log('routes');
      return <Routes />;
    }
  }
}
