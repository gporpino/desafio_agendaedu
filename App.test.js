import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('<App />', () => {
  it('has only 1 child', () => {
    const tree = shallow(<App />);
    expect(tree.children.length).toBe(1);
    expect(tree).toMatchSnapshot();
  });
});
