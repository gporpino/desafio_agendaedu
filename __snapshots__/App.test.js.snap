// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has only 1 child 1`] = `
<AppLoading
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;
